client/human/HumanClientFSM.h

```diff
@@ -372,10 +375,9 @@ struct QuittingGame : boost::statechart::state<QuittingGame, HumanClientFSM> {
 
     std::chrono::steady_clock::time_point m_start_time;
 
-    // if m_reset_to_intro is true QuittingGame transits to the Intro menu, otherwise it
-    // exits the app.
-    bool m_reset_to_intro;
-    Process* m_server_process;
+    Process* m_server_process = nullptr;
+    /// An action to be completed after disconnecting from the server
+    std::function<void()> m_after_server_shutdown_action = std::function<void()>();
```

Comment:
for clarity, or is explicitly calling the default construction doing something special here...?

Reply:
It was for clarity.  `m_server_process` needs a default value. `m_start_time` can't have a default value, it needs to be initialized to `Clock::now()`.  It was to be clear that `m_after_server_shutdown_action` was properly considered.